apps/server/src/modules/oauth/service/oauth.service.ts

```diff
@@ -167,55 +176,60 @@ export class OAuthService {
 		}
 		const oauthConfig: OauthConfig = this.extractOauthConfigFromSystem(system);
 
-		const migrationRedirect: string = this.userMigrationService.getMigrationRedirectUri(targetSystemId);
+		const migrationRedirect: string = this.userMigrationService.getMigrationRedirectUri();
 		const queryToken: OauthTokenResponse = await this.requestToken(authCode, oauthConfig, migrationRedirect);
 
 		await this.validateToken(queryToken.id_token, oauthConfig);
 
 		return queryToken;
 	}
 
-	/**
-	 * Builds the URL from the given parameters.
-	 *
-	 * @param provider
-	 * @param idToken
-	 * @param logoutEndpoint
-	 * @return built redirectUrl
-	 */
-	getRedirectUrl(provider: string, idToken = '', logoutEndpoint = ''): string {
-		const HOST = Configuration.get('HOST') as string;
+	getPostLoginRedirectUrl(provider: string, idToken = '', logoutEndpoint = '', postLoginRedirect?: string): string {
+		const clientUrl: string = Configuration.get('HOST') as string;
+		const dashboardUrl: URL = new URL('/dashboard', clientUrl);
```

Comment:
That is not good id hard code "/dashboard" if that moved for example to "/rooms"

And in general, if we had used this API for the APP, it would also be the task of the APP to decide where to go for navigation

Reply:
We know and it is unfortunate. We talked with @Metauriel about it and will move all redirects to the Client in Ticket https://ticketsystem.dbildungscloud.de/browse/N21-632